Identify builds git tagged with `prototype-...` in CLI and analytics

dureuill opened the pull request:
# Pull Request

## What does this PR do?

- Parses the last git tag to extract a prototype name if:
  - Current build uses the prototype tag (not after the tag) precisely
  - The prototype tag name respects the following conditions:
    1. starts with `prototype-`
    2. ends with a number
    3. the hyphen-separated segment right before the number is not a number (required to reject commits after the tag).
- Display the prototype name in the launch summary in the CLI
- Send the prototype name to analytics if any
- Update prototypes instructions in CONTRIBUTING.md

|`VERGEN_GIT_SEMVER_LIGHTWEIGHT` value | Prototype |
|---|---|
| `Some("prototype-geo-bounding-box-0-139-gcde89018")` | `None` (does not end with a number) |
| `Some("prototype-geo-bounding-box-0-139-89018")` | `None` (before the last segment is a number) |
| `Some("prototype-geo-bounding-box-0")` | `Some("prototype-geo-bounding-box-0")` |
| `Some("prototype-geo-bounding-box")` | `None` (does not end with a number") |
| `Some("geo-bounding-box-0")` | `None` (does not start with "prototype") |
| `None` | `None` |

irevoire reviewed (dismissed):
Good to me, you can merge when you want :giraffe:

dureuill commented:
Thank you :heart:, I'll do that as soon as the prototypes I launched are done, so sometimes tonight if everything goes well :-)

dureuill commented:
Updated:

- No longer checks for the dirtiness of the work directory. This should allow docker builds to recognize the prototypes
- Updated CONTRIBUTING.md, so adding @curquiza as reviewer

curquiza commented on CONTRIBUTING.md:
```diff
@@ -121,15 +121,20 @@ The full Meilisearch release process is described in [this guide](https://github
 
 The prototype name must follow this convention: `prototype-X-Y` where
-- `X` is the feature name formatted in `kebab-case`
+- `X` is the feature name formatted in `kebab-case`. It should not end with a number segment.
```
with a number segment?

dureuill replied:
yes, a component of the `kebab-cased` name that only contains something that can be parsed as a number.

There's an example below.

dureuill replied:
nah i don't think this has to do with not knowing the phrase, it is a bit confusing indeed.

Do you have a suggestion on how I could better express this?

curquiza replied:
Hum... 🤔 "It should end with a single number" -> not exactly true, but with the example below, we would understand `0-0` will not work

curquiza commented on CONTRIBUTING.md:
```diff
@@ -121,15 +121,20 @@ The full Meilisearch release process is described in [this guide](https://github
-Example: `prototype-auto-resize-0`.
+✅ Example: `prototype-auto-resize-0`.
+
+❌ Bad example: `auto-resize-0`: lacks the `prototype` prefix.
+❌ Bad example: `prototype-auto-resize`: lacks the version suffix.
+❌ Bad example: `prototype-auto-resize-0-0`: feature name ends with a number segment.
```
I think you want to put `<br>`

<img width="1096" alt="Capture d’écran 2023-02-09 à 17 58 24" src="https://user-images.githubusercontent.com/20380692/217884453-24413990-f40e-46e6-a93c-30eb7199fb57.png">

dureuill replied:
Added the brs

dureuill commented:
Looks like the docker actually doesn't build in a git workdir at all, but passes the variables manually.

To accommodate for this, I changed the Dockerfile and publish workflow.

curquiza reviewed (commented):
Ok for the CONTRIBUTING.md, but we need a developer approval 😄 
Thanks Louis for the work!

irevoire reviewed (dismissed):
🐰

dureuill commented:
Confirmed working on docker with `prototype-unlimited-indexes-1`!

bors merge

curquiza reviewed (approved):
bors merge
